config/config.go

```diff
@@ -149,12 +141,21 @@ func loadConfig() (*config, error) {
 		configFileError = err
 	}
 
+	// Warn about missing Config file only after all other configuration is
+	// done.  This prevents the warning on help messages and invalid
+	// options.  Note this should go directly before the return.
+	if configFileError != nil {
+		log.Warning("%v", configFileError)
+	}
+
+	// TODO: move out of here
 	// Finally, parse the remaining command line options again to ensure
 	// they take precedence.
-	if _, err := flags.Parse(&cfg); err != nil {
-		return nil, err
-	}
+	return ParseFlags(cfg)
+}
 
+// SetupConfig initializes filesystem and network infrastructure.
+func SetupConfig(cfg *Config) (*Config, error) {
```

Comment:
The logging in this function is now incorrect. It logs as `loadConfig` instead of as `SetupConfig`.

Also the error handling in general is outdated; instead of casting the `err` variable, an `errors.As` might be a more modern approach.